src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -381,7 +379,7 @@ stepWithRule
         proof = renamingProof <> unificationProof'
         attachProof result = (result, proof)
     results <-
-        keepGoodResults $ return $ map
+        traverse
```

Comment:
If not too difficult, I would add a test that failed before this change, but passes now.